### PR TITLE
Don't fork a thread for stdin if not necessary

### PR DESCRIPTION
**Pull request by soenkehahn**

### CODE REVIEW

File: src/collected_output.rs [hunk 1]
```diff
@@ -52,11 +52,13 @@ impl Waiter {
         Stdout: Write + Send + Clone + 'static,
         Stderr: Write + Send + Clone + 'static,
     {
-        let config_stdin = config.stdin.clone();
-        let stdin_join_handle = thread::spawn(move || -> io::Result<()> {
-            child_stdin.write_all(&config_stdin)?;
-            Ok(())
-        });
+        let stdin_join_handle = match config.stdin.clone() {
```
Comment (casey) [inline src/collected_output.rs:55]: It looks like this can be done with `.map`. TIOLI.

**Comment by casey** [comment]
LGTM!

File: src/collected_output.rs (hunk 1 above)
Comment (soenkehahn) [reply src/collected_output.rs:55]: Yeah, I actually tried that out but thought that it obscured a bit what was going on.

Comment (soenkehahn) [reply src/collected_output.rs:55]: Some older versions of `clippy` (e.g. in rustc `1.53`) also complain about this. So I'm switching to `.map`. If you *and* `clippy` say so... :shrug:

**Pull request merged**